appserver/jdbc/jdbc-ra/jdbc-core/src/main/java/com/sun/gjc/util/MethodExecutor.java

```diff
@@ -87,28 +89,38 @@ public void runJavaBeanMethod(String value, Method method, Object obj) throws Re
         if (value == null || value.trim().equals("")) {
             return;
         }
-        try {
-            Class[] parameters = method.getParameterTypes();
-            if (parameters.length == 1) {
-                Object[] values = new Object[1];
-                values[0] = convertType(parameters[0], value);
-                method.invoke(obj, values);
+
+        Class[] parameters = method.getParameterTypes();
+        if (parameters.length == 1) {
+            Object[] values = new Object[1];
+            values[0] = convertType(parameters[0], value);
+
+            final ResourceException[] exception = new ResourceException[1];
+            AccessController
+                .doPrivileged(new PrivilegedAction() {
```

Comment:
Seeing the way you are capturing the exception, I'm wondering why you didn't use PrivilegedExceptionAction since the code within run() is throwing checked exceptions?